prov/efa/src/rdm/efa_rdm_ep_fiops.c

```diff
@@ -1175,6 +1175,35 @@ static int efa_rdm_ep_set_cuda_api_permitted(struct efa_rdm_ep *ep, bool cuda_ap
 	return 0;
 }
 
+/**
+ * @brief act on shared_memory_permitted flag called by efa_rdm_ep_setopt
+ * @param[in,out]	ep			endpoint
+ * @param[in]		shm_permitted	whether shared memory is permitted
+ * @return		0 on success,
+ *			-FI_EINVAL if shm is requested but the FI_EFA_ENABLE_SHM_TRANSFER environment variable is set to false
+ * @related efa_rdm_ep
+ */
+static int efa_rdm_ep_set_shared_memory_permitted(struct efa_rdm_ep *ep, bool shm_permitted)
+{
+	if (!shm_permitted) {
+		EFA_WARN(FI_LOG_EP_CTRL,
+			 "FI_OPT_SHARED_MEMORY_PERMITTED set to false. "
+			 "Tearing down shm provider resources.");
+		efa_rdm_ep_close_shm_resources(ep);
```

Comment:
We have another place that close shm resource during ep enable: https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/rdm/efa_rdm_ep_fiops.c#L979. I think it's better to keep the logic consistent: tear down shm inside setopt, or both clean up during ep enable. I think the first one is better (as you do now). Can you update the setopt for cuda_api_permitted to do the same?

Reply:
Then we can remove the `efa_rdm_ep_update_shm` function

Reply:
I will move the teardown in cuda_api_permitted to setopt in a separate commit

Reply:
I see. Then I will move all cleanup to ep enable.